xpublish_edr/geometry/position.py

```diff
@@ -28,3 +41,17 @@ def _select_by_position_regular_xy_grid(
     """
     # Find the nearest X and Y coordinates to the point
     return ds.cf.sel(X=point.x, Y=point.y, method="nearest")
+
+
+def _select_by_multiple_positions_regular_xy_grid(
+    ds: xr.Dataset,
+    points: shapely.MultiPoint,
+) -> xr.Dataset:
+    """
+    Return a dataset with the positions nearest to the given coordinates
+    """
+    # Find the nearest X and Y coordinates to the point using vectorized indexing
+    x, y = np.array(list(zip(*[(point.x, point.y) for point in points.geoms])))
+    sel_x = xr.Variable(data=x, dims=VECTORIZED_DIM)
+    sel_y = xr.Variable(data=y, dims=VECTORIZED_DIM)
+    return ds.cf.sel(X=sel_x, Y=sel_y, method="nearest")
```

Comment:
Don't we want something like this?

```suggestion
    if method=="nearest":
        return ds.cf.sel(X=sel_x, Y=sel_y, method="nearest")
    elif method="bilinear':
        return ds.cf.interp(X=sel_x, Y=sel_y)
```